src/extractor.rs

```diff
@@ -418,30 +468,57 @@ fn type_check(ast: ExprU) -> Result<ExprT, TypeError> {
             };
             match &name[..] {
                 "ref" => {
-                    if args.len() != 1 && args.len() != 2 {
-                        return Err(TypeError::ArgumentMismatch {
-                            expected: vec![1, 2],
-                            found: args.len(),
-                        });
-                    }
-                    let typed_args = args
-                        .into_iter()
-                        .map(|arg| {
-                            match arg {
-                                // refs can only take string literals
-                                ExprU::StringU(s) => Ok(s),
-                                // error on everything that isn't a string literal
-                                not_string => Err(TypeError::TypeMismatch {
-                                    expected: ExprType::String,
-                                    got: ExprType::from(&not_string),
-                                }),
-                            }
-                        })
-                        .collect::<Result<Vec<String>, TypeError>>()?;
-                    Ok(ExprT::RefT(
-                        typed_args[0].clone(),
-                        typed_args.get(1).map(|x| x.to_owned()),
-                    ))
+                    let mut arg_stack = Vec::from(args);
```

Comment:
`args` is already a `Vec` so we don't need the `from` conversion.

Reply:
if you're looking assign it to satisfy the borrow checker, use `clone()` instead.